app/src/main/java/com/arygm/quickfix/ui/elements/ScrollableRow.kt

```diff
@@ -0,0 +1,132 @@
+                  color = MaterialTheme.colorScheme.onBackground,
+                  textAlign = TextAlign.Center,
+                  style = MaterialTheme.typography.headlineMedium,
+                  maxLines = 1,
+                  modifier =
+                      Modifier.fillMaxWidth()
+                          .weight(0.3f) // Allocate 30% of card height to the text
+                          .wrapContentHeight(Alignment.CenterVertically)
+                          .testTag("ServiceName_${service.name}"))
+            }
+      }
+}
+
+// PopularServicesRow composable with percentage-based card width and consistent padding
+@Composable
+fun PopularServicesRow(
+    services: List<Service>,
+    onServiceClick: (Service) -> Unit,
+    modifier: Modifier = Modifier
+) {
+  BoxWithConstraints {
+    val cardWidth = maxWidth * 0.4f // 40% of the available width for each card
+    val horizontalSpacing = maxWidth * 0.025f // 2.5% of the available width for spacing
+
+    LazyRow(
+        modifier = modifier.fillMaxWidth().testTag("PopularServicesRow"),
+        horizontalArrangement = Arrangement.spacedBy(horizontalSpacing),
+        contentPadding = PaddingValues(start = horizontalSpacing, end = horizontalSpacing)) {
+          items(services.size) { index ->
+            ServiceCard(service = services[index], modifier = Modifier.width(cardWidth)) {
+              onServiceClick(services[index])
+            }
+          }
+        }
+  }
+}
+
+// MainContent composable (the screen that contains all sections)
+@Composable
+fun MainContent() {
+  val backgroundColor = MaterialTheme.colorScheme.background
+
+  val services =
+      listOf(
+          Service("Mechanic", R.drawable.mechanic),
+          Service("Gardener", R.drawable.gardener),
+          Service("Electrician", R.drawable.electrician),
+          Service("Painter", R.drawable.painter),
+          Service("Plumber", R.drawable.plumber),
+      )
+
+  Column(
+      modifier =
+          Modifier.fillMaxSize()
+              .background(backgroundColor) // Consistent background color
+              .testTag("MainContent")) {
+        Text(
+            text = "Popular services",
+            color = MaterialTheme.colorScheme.onBackground,
+            style = MaterialTheme.typography.headlineMedium,
+            modifier =
+                Modifier.padding(start = 15.dp).testTag("PopularServicesTitle").fillMaxWidth())
+        PopularServicesRow(
+            services = services,
+            onServiceClick = { service ->
+              // Action when a service is clicked
+            },
+            modifier = Modifier)
+      }
+}
+
+// Preview for MainContent to ensure consistent styling
+@Preview(showBackground = true)
+@Composable
+fun PreviewMainContent() {
+  QuickFixTheme {
+    Surface(
+        modifier = Modifier.fillMaxSize(),
+        color = MaterialTheme.colorScheme.background // Consistent background color
+        ) {
+          MainContent()
+        }
+  }
+}
```

Comment:
This should be removed as it does not serve any purpose other than visual testing

Reply:
I’d like to keep it as it is for now since this isn't the final version. I'm waiting for the backend integration to complete the final version.